src/VCR/Util/CurlHelper.php

```diff
@@ -175,24 +175,39 @@ public static function setCurlOptionOnRequest(Request $request, $option, $value,
             case CURLOPT_HEADER:
             case CURLOPT_WRITEFUNCTION:
             case CURLOPT_HEADERFUNCTION:
+            case CURLOPT_UPLOAD:
                 // Ignore header, file and writer functions.
                 // These options are stored and will be handled later in handleOutput().
                 break;
-            case CURLOPT_READFUNCTION:
```

Comment:
Removing handling the curl option here was probably necessary because it is not clear if `CURLOPT_POSTFIELDS` have been set yet, right? As far as I understand this came from a special case when using Guzzle 4. Do you think it is possible to leave it here and check for the curl option because Guzzle 4 sets it before the `CURLOPT_READFUNCTION` option?